api/tests/test_submit_job.py

```diff
@@ -99,3 +99,46 @@ def test_submit_job_granule_does_not_exist(client, table):
     assert response.json['title'] == 'Bad Request'
     assert response.json['detail'] == 'Some requested scenes could not be found: ' \
                                       'S1A_IW_SLC__1SDV_20200610T173646_20200610T173704_032958_03D14C_5F2A'
+
+
+def test_submit_good_granule_names(client, table):
+    login(client)
+    good_granule_names = [
+        'S1B_IW_SLC__1SDV_20200604T082207_20200604T082234_021881_029874_5E38',
+        'S1A_IW_SLC__1SSH_20150608T205059_20150608T205126_006287_0083E8_C4F0',
+        'S1A_IW_GRDH_1SDV_20200604T190627_20200604T190652_032871_03CEB7_56F3',
+        'S1A_IW_GRDH_1SSH_20171122T184459_20171122T184524_019381_020DD8_B825',
+    ]
+    for granule in good_granule_names:
+        batch = [
+            make_job(granule),
+        ]
+        setup_requests_mock(batch)
+        response = submit_batch(client, batch)
+        assert response.status_code == status.HTTP_200_OK
+
+
+def test_submit_bad_granule_names(client):
+    login(client)
+    bad_granule_names = [
+        'foo',
+        'S1B_IW_SLC__1SDV_20200604T082207_20200604T082234_021881_029874_5E3',
+        'S1B_IW_SLC__1SDV_20200604T082207_20200604T082234_021881_029874_5E38_',
```

Comment:
Might want an `S1C` bad granule as well -- nothing is checking the first part of the regex currently